test/distributed/tensor/test_view_ops.py

```diff
@@ -242,6 +242,41 @@ def test_illegal_views(self):
         ):
             shard.view(-1)
 
+    @with_comms
+    def test_flatten_adjacent(self):
+        mesh_shape = (dist.get_world_size() // 2, 2)
+        device_mesh = init_device_mesh(
+            self.device_type, mesh_shape=mesh_shape, mesh_dim_names=("outer", "inner")
+        )
+
+        # Special case where flattening 2 sharded dims can be supported without communication
+        tensor = torch.randn((8, 256))
+        dtensor = distribute_tensor(tensor, device_mesh, [Shard(0), Shard(1)])
```

Comment:
a general description could be: the DTensor is "contiguous". Maybe we can come up with a more formal definition.